File: src/main/java/com/trade/securities/application/service/StockDataService.java
package com.trade.securities.application.service;

import com.trade.securities.application.port.in.LoadStockDataUseCase;
import com.trade.securities.application.port.out.LoadMarketDataPort;
import com.trade.securities.domain.StockData;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class StockDataService implements LoadStockDataUseCase {

    private final LoadMarketDataPort loadMarketDataPort;

    @Override
    public List<StockData> getStockData(String date, String type) {
        return loadMarketDataPort.loadStockData(date, type);
    }
}

File: src/main/kotlin/com/trade/securities/application/service/StockDataService.kt
package com.trade.securities.application.service

import com.trade.securities.application.port.`in`.LoadStockDataUseCase
import com.trade.securities.application.port.out.LoadMarketDataPort
import com.trade.securities.domain.StockData
import org.springframework.stereotype.Service

@Service
class StockDataService(
    private val loadMarketDataPort: LoadMarketDataPort
) : LoadStockDataUseCase {

    override fun getStockData(date: String, type: String): List<StockData> {
        return loadMarketDataPort.loadStockData(date, type)
    }
}
